gw_eccentricity/exceptions.py

```diff
@@ -0,0 +1,58 @@
+"""Custom exception classes for gw_eccentricity."""
+
+
+class InsufficientExtrema(Exception):
+    """Exception raised when number of extrema is not enough.
+
+    Parameters
+    ----------
+    extrema_type : str
+        Type of extrema. Can be "pericenter" or "apocenter".
+    num_extrema : int
+        Number of extrema.
+    additional_message : str
+        Any additional message to append to the exception message.
+        Default is None which adds no additional message.
+    """
+
+    def __init__(self, extrema_type, num_extrema, additional_message=None):
+        """Init for InsufficientExtrema Class."""
+        self.extrema_type = extrema_type
+        self.num_extrema = num_extrema
+        self.additional_message = additional_message
+        self.message = (f"Number of {self.extrema_type} is {self.num_extrema}."
+                        f" Number of {self.extrema_type} is not sufficient "
+                        f"to build frequency interpolant.")
+        if self.additional_message is not None:
+            self.message += "\n" + self.additional_message
+        super().__init__(self.message)
+
+
+class NotInRange(Exception):
```

Comment:
Same, why does this exist?
Also, name is too generic...

Reply:
Similarly, it helps identify the exception cause when using try and except. I have modified the name and added documentation.